app/models/user.rb

```diff
@@ -77,6 +77,7 @@ class User < ActiveRecord::Base
   has_many :topics_allowed, through: :topic_allowed_users, source: :topic
   has_many :groups, through: :group_users
   has_many :secure_categories, through: :groups, source: :categories
+  has_many :invites, through: :invited_users
```

Comment:
This is a breaking change because we're no longer deleting invites when the user is deleted.

Reply:
But we don't really want to delete an invite in case the invite is an invite link which can be redeemed by multiple users.

Reply:
What about the previous case where a user sends an invite to a particular email? We used to delete those when the user is deleted but we don't do that anymore now.

Reply:
That is true. What do you suggest here?

Reply:
Actually 

> But we don't really want to delete an invite in case the invite is an invite link which can be redeemed by multiple users.

Why do we not want to delete an invite link which can be redeemed by multiple users? If the user that created the link has been deleted, the any invite links which has been created by the user should not be valid.

Reply:
The `has_many` association in `users` table refers to `user_id` column in `invites` table which <s>is</s> was the ID of the _invited_ user. So it seems like when the user is deleted the corresponding invite record which was redeemed is deleted. This has always been the case.

Now when a user is deleted we'll delete the corresponding `invited_users` record.

Reply:
Thank you for clarifying. Let me try to explain where I'm confused.

1. As a staff user, I can generate an invite link. That invite link is created as an `Invite` record with the `invited_by_id` set to the staff user. When the staff user is deleted, we should delete the invite record so that the link is no longer valid.

1. On the `User` model, we have `has_many :invited_users, dependent: :delete_all`. This means `SELECT * FROM invited_users WHERE invited_users.user_id = <current user id>`. The `InvitedUser` table as I understand is to maintain the link between the invite and the user that was invited through that invite. A user can only be invited once so how can a user `has_many :invited_users`?

1. `has_many :invites, through: :invited_users`. Since we're going through the `invited_users` table, this is looking for the invite that the current user was invited from which I think is wrong. Instead shouldn't a user's relationship to `invites` be `has_many :invites, foreign_key: :invited_by_id`?

Reply:
I think you're correct. We should link invites that user has created, not redeemed. I'll make the required changes.

Reply:
Done.